eta/core/annotations.py

```diff
@@ -581,14 +675,16 @@ def _render_attr_name_value(attr, show_confidence=True):
     return attr_str
 
 
-def _render_object_label(obj, show_confidence=True):
+def _render_object_label(obj, show_index=True, show_confidence=True):
+    add_confidence = show_confidence and obj.confidence is not None
+    add_index = show_index and obj.index is not None
+
     label_str = _clean(obj.label).upper()
 
-    add_confidence = show_confidence and obj.confidence is not None
     if add_confidence:
         label_str += " (%.2f)"
 
-    if obj.index is not None:
+    if add_index:
```

Comment:
`if add_index and obj.index is not None`

Reply:
`add_index` already includes that check